core/templates/pages/error-pages/error-page.component.spec.ts

```diff
@@ -15,114 +15,37 @@
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+import { TranslateModule } from '@ngx-translate/core';
 
 import { ErrorPageComponent } from './error-page.component';
-import { UrlInterpolationService } from
-  'domain/utilities/url-interpolation.service';
-import { I18nLanguageCodeService } from 'services/i18n-language-code.service';
-import { PageTitleService } from 'services/page-title.service';
-import { MockTranslatePipe } from 'tests/unit-test-utils';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
 
-class MockI18nLanguageCodeService {
-  codeChangeEventEmiiter = new EventEmitter<string>();
-  getCurrentI18nLanguageCode() {
-    return 'en';
-  }
+describe('ErrorPageComponent', () => {
+  let component: ErrorPageComponent;
+  let fixture: ComponentFixture<ErrorPageComponent>;
 
-  get onI18nLanguageCodeChange() {
-    return this.codeChangeEventEmiiter;
-  }
-}
-
-class MockTranslateService {
-  onLangChange: EventEmitter<string> = new EventEmitter();
-  instant(key: string, interpolateParams?: Object): string {
-    return key;
-  }
-}
-
-let component: ErrorPageComponent;
-let fixture: ComponentFixture<ErrorPageComponent>;
-
-describe('Error page', () => {
-  let pageTitleService: PageTitleService;
-  let translateService: TranslateService;
-
-  beforeEach(async(() => {
+  beforeEach(() => {
     TestBed.configureTestingModule({
-      declarations: [ErrorPageComponent, MockTranslatePipe],
-      providers: [
-        {
-          provide: I18nLanguageCodeService,
-          useClass: MockI18nLanguageCodeService
-        },
-        UrlInterpolationService,
-        PageTitleService,
-        {
-          provide: TranslateService,
-          useClass: MockTranslateService
-        }
-      ],
-      schemas: [NO_ERRORS_SCHEMA]
+      imports: [TranslateModule.forRoot()],
+      declarations: [ErrorPageComponent],
+      providers: [UrlInterpolationService],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA]
     }).compileComponents();
-  }));
 
-  beforeEach(() => {
     fixture = TestBed.createComponent(ErrorPageComponent);
     component = fixture.componentInstance;
-    pageTitleService = TestBed.inject(PageTitleService);
-    translateService = TestBed.inject(TranslateService);
     component.statusCode = '404';
-    fixture.detectChanges();
   });
 
   it('should check if status code is a number', () => {
     expect(component.getStatusCode()).toBe(404);
     expect(component.getStatusCode()).toBeInstanceOf(Number);
   });
 
-  it('should set images and subscribe to onLangChange upon initialization',
-    () => {
-      spyOn(translateService.onLangChange, 'subscribe');
-      component.ngOnInit();
-      expect(component.getStaticImageUrl('/general/oops_mint.webp'))
-        .toBe('/assets/images/general/oops_mint.webp');
-      expect(translateService.onLangChange.subscribe).toHaveBeenCalled();
-    });
-
-  it('should obtain translated page title whenever the selected' +
```

Comment:
Why do we delete all these tests?

Reply:
The error page was setting the page title and also subscribing to the translate service. All of this is now done in the Base Root component that is used by the Error Root component. So I removed this bit of code from the error page component and also removed corresponding tests.